libraries/AP_Math/examples/throttle_curve/throttle_curve.cpp

```diff
@@ -0,0 +1,108 @@
+//
+// Unit tests for the AP_Math throttle curve code
+//
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+
+void test_throttle_curve(float alpha);
+void setup();
+void loop();
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
```

Comment:
Is there anything in here which requires SITL?

Reply:
just doesn't make sense to run it anywhere else

Reply:
Why does it need simulated sensor drivers?  Why does it need a state machine for handling program state?

AFAICS this should be a linux native binary.  Or a Windows one, for that matter.

Reply:
The key thing I wanted to achieve was to call AP_Math::throttle_curve directly, and failing that to copy/paste the code_under_test. 
The first iteration of this used a custom makefile to accomplish that, but doing it this way eliminated the need for that file and instructions on how to build it. It's becoming very obvious that, while this accomplishes my goal of plotting a curve without transliterating C++ to python, there must be a better way to meet those 2 objectives :)